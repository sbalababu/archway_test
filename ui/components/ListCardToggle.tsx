import * as React from 'react';
import { Icon } from 'antd';
import Colors from './Colors';

interface Props {
  selectedMode: string;
  style?: any;
  onSelect?: (mode: string) => void;
}

const toggles = [{ text: 'list', icon: 'bars' }, { text: 'cards', icon: 'appstore' }];

const ListCardToggle = ({ selectedMode, style = {}, onSelect }: Props) => {
  return (
    <div
      style={{
        display: 'flex',
        justifyContent: 'flex-end',
        ...style,
      }}
    >
      {toggles.map(({ text, icon }, index) => (
        <div
          key={text}
          style={{
            textAlign: 'center',
            width: 36,
            color: Colors.PrimaryColor.string(),
            borderBottomWidth: 1,
            borderBottomStyle: 'solid',
            borderBottomColor: selectedMode === text ? Colors.PrimaryColor.string() : 'transparent',
            paddingBottom: 4,
            margin: '0 4px',
            cursor: 'pointer',
          }}
          onClick={() => !!onSelect && onSelect(text)}
        >
          <Icon type={icon} style={{ fontSize: '20px' }} />
          <span
            style={{
              display: 'block',
              fontSize: '10px',
              lineHeight: '10px',
              textTransform: 'uppercase',
            }}
          >
            {text}
          </span>
        </div>
      ))}
    </div>
  );
};

export default ListCardToggle;
