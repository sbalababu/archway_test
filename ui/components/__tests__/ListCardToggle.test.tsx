import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ListCardToggle from '../ListCardToggle';

describe('ListCardToggle', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ListCardToggle selectedMode="cards" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
