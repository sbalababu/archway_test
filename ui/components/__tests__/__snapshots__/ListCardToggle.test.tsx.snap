// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCardToggle renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "flex-end",
    }
  }
>
  <div
    key="list"
    onClick={[Function]}
    style={
      Object {
        "borderBottomColor": "transparent",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "color": "rgb(45, 116, 147)",
        "cursor": "pointer",
        "margin": "0 4px",
        "paddingBottom": 4,
        "textAlign": "center",
        "width": 36,
      }
    }
  >
    <Icon
      style={
        Object {
          "fontSize": "20px",
        }
      }
      type="bars"
    />
    <span
      style={
        Object {
          "display": "block",
          "fontSize": "10px",
          "lineHeight": "10px",
          "textTransform": "uppercase",
        }
      }
    >
      list
    </span>
  </div>
  <div
    key="cards"
    onClick={[Function]}
    style={
      Object {
        "borderBottomColor": "rgb(45, 116, 147)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "color": "rgb(45, 116, 147)",
        "cursor": "pointer",
        "margin": "0 4px",
        "paddingBottom": 4,
        "textAlign": "center",
        "width": 36,
      }
    }
  >
    <Icon
      style={
        Object {
          "fontSize": "20px",
        }
      }
      type="appstore"
    />
    <span
      style={
        Object {
          "display": "block",
          "fontSize": "10px",
          "lineHeight": "10px",
          "textTransform": "uppercase",
        }
      }
    >
      cards
    </span>
  </div>
</div>
`;
